dash/_patch.py

```diff
@@ -0,0 +1,117 @@
+def _operation(name, location, **kwargs):
+    return {"operation": name, "location": location, "params": dict(**kwargs)}
+
+
+_noop = object()
+
+
+class Patch:
+    """
+    Patch a callback output value
+
+    Act like a proxy of the output prop value on the frontend.
+
+    Supported prop types: Dictionaries and lists.
+    """
+
+    def __init__(self, location=None, parent=None):
+        if location is not None:
+            self._location = location
+        else:
+            # pylint: disable=consider-using-ternary
+            self._location = (parent and parent._location) or []
+        if parent is not None:
+            self._operations = parent._operations
+        else:
+            self._operations = []
+
+    def __getitem__(self, item):
+        return Patch(location=self._location + [item], parent=self)
+
+    def __getattr__(self, item):
+        if item == "tolist":
+            # to_json fix
+            raise AttributeError
+        if item == "_location":
+            return self._location
+        if item == "_operations":
+            return self._operations
+        return self.__getitem__(item)
+
+    def __setattr__(self, key, value):
+        if key in ("_location", "_operations"):
+            self.__dict__[key] = value
+        else:
+            self.__setitem__(key, value)
+
+    def __delattr__(self, item):
+        self.__delitem__(item)
+
+    def __setitem__(self, key, value):
+        if value is _noop:
+            # The += set themselves.
+            return
+        self._operations.append(
+            _operation(
+                "Assign",
+                self._location + [key],
+                value=value,
+            )
+        )
+
+    def __delitem__(self, key):
+        self._operations.append(_operation("Delete", self._location + [key]))
+
+    def __add__(self, other):
+        self._operations.append(_operation("Add", self._location, value=other))
+        return _noop
+
+    def __iadd__(self, other):
+        self._operations.append(_operation("Add", self._location, value=other))
+        return _noop
+
+    def __sub__(self, other):
+        self._operations.append(_operation("Sub", self._location, value=other))
+        return _noop
+
+    def __isub__(self, other):
+        self._operations.append(_operation("Sub", self._location, value=other))
+        return _noop
+
+    def __mul__(self, other):
+        self._operations.append(_operation("Mul", self._location, value=other))
+        return _noop
+
+    def __imul__(self, other):
+        self._operations.append(_operation("Mul", self._location, value=other))
```

Comment:
In principle I guess list multiplication is also a thing... I can't see that being useful though.